autofit/non_linear/optimize/lbfgs.py

```diff
@@ -0,0 +1,251 @@
+
+            if iterations > 0:
+
+                config_dict_options = self.config_dict_options
+                config_dict_options["maxiter"] = iterations
+
+                lbfgs = optimize.minimize(
+                    fun=fitness_function.__call__,
+                    x0=x0,
+                    method="L-BFGS-B",
+                    options=config_dict_options,
+                    **self.config_dict_search
+                )
+
+                total_iterations += lbfgs.nit
+
+                self.paths.save_object(
+                    "total_iterations",
+                    total_iterations
+                )
+                self.paths.save_object(
+                    "log_posterior",
+                    fitness_function.log_posterior_from(parameter_list=lbfgs.x)
+                )
+                self.paths.save_object(
+                    "x0",
+                    lbfgs.x
+                )
+
+                self.perform_update(
+                    model=model, analysis=analysis, during_analysis=True
+                )
+
+                x0 = lbfgs.x
+
+                if lbfgs.nit < iterations:
+                    return
+
+        self.logger.info("L-BFGS sampling complete.")
+
+    def samples_from(self, model):
+
+        return LBFGSSamples(
+            model=model,
+            x0=self.paths.load_object("x0"),
+            log_posterior_list=np.array([self.paths.load_object("log_posterior")]),
+            total_iterations=self.paths.load_object("total_iterations"),
+            time=self.timer.time
+        )
+
+
+class LBFGSSamples(OptimizerSamples):
+
+    def __init__(
+            self,
+            model: AbstractPriorModel,
+            x0: np.ndarray,
+            log_posterior_list: np.ndarray,
+            total_iterations: int,
+            time: Optional[float] = None,
+    ):
+        """
+        Create an *OptimizerSamples* object from this non-linear search's output files on the hard-disk and model.
+
+        For PySwarms, all quantities are extracted via pickled states of the particle and cost histories.
+
+        Parameters
+        ----------
+        model
+            The model which generates instances for different points in parameter space. This maps the points from unit
+            cube values to physical values via the priors.
+        """
+
+        self.x0 = x0
+        self._log_posterior_list = log_posterior_list
+        self.total_iterations = total_iterations
+
+        super().__init__(
+            model=model,
+            time=time,
+        )
+
+        self._samples = None
+
+    @property
+    def samples(self):
+        """
+        Create a `Samples` object from this non-linear search's output files on the hard-disk and model.
+
+        For L-BFGS, only a single ndarray of parameters is returned, alongside its log posterior value. Thus, the
+        `LBFGSSamples` object contains only one entry.
+        """
+
+        if self._samples is not None:

```

Comment:
You can get rid of one whole line of code by checking inverting this logic and using only one return statement!